src/lib/sticky-header/sticky-header.spec.ts

```diff
@@ -0,0 +1,180 @@
+      .getBoundingClientRect().top;
+    expect(stickyHeader.element.getBoundingClientRect().top).not.toBe(scrollableContainerTop);
+    tick();
+
+    // Scroll the scrollableContainer up to stick
+    fixture.componentInstance.scrollDown();
+    // Use tick(100) to tick forward to let stickyHeader's _applyStickyPositionStyles()
+    // function finished
+    tick(100);
+
+    expect(stickyHeader.element.getBoundingClientRect().top).toBe(scrollableContainerTop);
+  }));
+
+  it('should unstuck when scrolled off the top of the container', fakeAsync(() => {
+    let scrollableContainerTop = stickyHeader.upperScrollableContainer
+      .getBoundingClientRect().top;
+    expect(stickyHeader.element.getBoundingClientRect().top).not.toBe(scrollableContainerTop);
+    tick();
+
+    // Scroll the scrollableContainer up to stick
+    fixture.componentInstance.scrollDown();
+    // Use tick(100) to tick forward to let stickyHeader's _applyStickyPositionStyles()
+    // function finished
+    tick(100);
+
+    expect(stickyHeader.element.getBoundingClientRect().top).toBe(scrollableContainerTop);
+
+    // Scroll the scrollableContainer down to unstuck
+    fixture.componentInstance.scrollBack();
+    tick(100);
+
+    expect(stickyHeader.element.getBoundingClientRect().top).not.toBe(scrollableContainerTop);
+
+  }));
+});
+
+describe('sticky-header with positioning supported', () => {
+  let fixture: ComponentFixture<StickyHeaderTest>;
+  let testComponent: StickyHeaderTest;
+  let stickyElement: DebugElement;
+  let stickyParentElement: DebugElement;
+  let stickyHeader: CdkStickyHeader;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [ OverlayModule, PlatformModule, StickyHeaderModule ],
+      declarations: [StickyHeaderTest],
+      providers: [
+        {provide: STICKY_HEADER_SUPPORT_STRATEGY, useValue: true},
+      ],
+    });
+    TestBed.compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(StickyHeaderTest);
+    fixture.detectChanges();
+    testComponent = fixture.debugElement.componentInstance;
+    stickyElement = fixture.debugElement.query(By.directive(CdkStickyHeader));
+    stickyParentElement = fixture.debugElement.query(By.directive(CdkStickyRegion));
+    stickyHeader = stickyElement.injector.get<CdkStickyHeader>(CdkStickyHeader);
+  });
+
+  it('should find sticky positioning is applied', () => {
+    let position = window.getComputedStyle(stickyHeader.element).position;
+    expect(position).not.toBe(null);
+    if (position != null) {
+      expect(/sticky/i.test(position)).toBe(true);
+    }
+  });
+});
+
+@Component({
+  // Use styles to define the style of scrollable container and header,
+  // which help test to make sure whether the header is stuck at the right position.
+  styles:[`
+    .scrollable-style {
+        text-align: center;
+        -webkit-appearance: none;
+        -moz-appearance: none;
+        height: 300px;
+        overflow: auto;
```

Comment:
-2 indent in the css (the properties are indented as +4 right now, they should be +2)